### PR TITLE
Add container for Github CI actions

### PR DESCRIPTION
**Pull request by aconsroe-hpe**
This adds a Dockerfile for use with our CI actions to reduce their
worst case-execution time by eliminating the need to download dependencies.

This is just phase 1 of 2, where we build the dockerfile and push it
to ghcr (github container registry). The second phase is to actually
use it in CI.yml -- I just don't know how to order these actions
correctly so I'm doing in two PRs

The container image should only get rebuilt when its Dockerfile
changes -- in other words, 99% of PRs shouldn't run this action

Links:
  * https://docs.github.com/en/actions/publishing-packages/publishing-docker-images
  * https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Signed-off-by: Andrew Consroe <andrew.consroe@hpe.com>

### CODE REVIEW

File: util/dockerfiles/github-ci/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# This is a container used to run CI jobs on GH
+FROM ubuntu:22.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+    bash \
+    clang-13 \
+    cmake \
+    cmake \
```
Comment (ronawho) [inline util/dockerfiles/github-ci/Dockerfile:10]: Can you add `cscope` too so we don't need to download that for `check_annotations_rt_calls`

**Comment by ronawho** [comment]
Can you add a link to the docs you used to build this? (either to the PR description or just a comment for future reference)

File: util/dockerfiles/github-ci/Dockerfile (hunk 1 above)
Comment (aconsroe-hpe) [reply util/dockerfiles/github-ci/Dockerfile]: Yep! Missed that, but luckily left myself a duplicate `cmake` which is in the perfect spot to change to `cscope`!

**Comment by aconsroe-hpe** [comment]
Does [this](https://github.com/chapel-lang/chapel/pull/19425/commits/1ce25b4c1f69455996d8d6547115c99ed0b4580f) look good @ronawho ?

**Comment by ronawho** [comment]
Sorry, I meant like the github action docs you based build-CI-container.yml off of (I'd like to know more for say doing something similar with Arkouda and figured it could be a useful reference if others need to modify ours in the future)

File: util/dockerfiles/github-ci/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This is a container used to run CI jobs on GH
+FROM ubuntu:22.04
```
Comment (ronawho) [inline util/dockerfiles/github-ci/Dockerfile:2]: Currently, we just use the `ubuntu-latest` alias to use github actions newest ubuntu container. Is there a way to do something similar here so we don't have to remember to bump our versions?

**Comment by aconsroe-hpe** [comment]
> Sorry, I meant like the github action docs you based build-CI-container.yml off of (I'd like to know more for say doing something similar with Arkouda and figured it could be a useful reference if others need to modify ours in the future)

Oh gotcha; I ended up needing to pull from various places, I will see if I can summarize those

File: util/dockerfiles/github-ci/Dockerfile (hunk 2 above)
Comment (aconsroe-hpe) [reply util/dockerfiles/github-ci/Dockerfile:2]: I opted for 22.04 to get llvm-13 available. If you do `FROM ubuntu:latest`, you'll get 20.04 today which only has llvm-12

Comment (ronawho) [reply util/dockerfiles/github-ci/Dockerfile:2]: Gotcha, can you add a comment along that lines.

**Review by ronawho — APPROVED** [review]

**Pull request merged**